### PR TITLE
[go] UI fixes and tweaks

### PR DESCRIPTION
**Pull request by Simek**
# Why

There are some slight inconsistencies across the Expo Go interface; let's address that.

# How

The following changes have been made:
* unify the default background color of screens
* Fix account modal appearance, tweak the iOS header
* fix spacing issues on the Home and Settings screens
* tweak the labels and wording around supported SDK
* add missing tab bar border

# Test Plan

The changes have been reviewed by running a debug build of Expo Go locally.

# Preview

https://github.com/expo/expo/assets/719641/39573985-84bd-4166-91b6-8737d4f746fd

### CODE REVIEW

File: apps/expo-go/src/screens/SettingsScreen/ConstantsSection.tsx [hunk 1]
```diff
@@ -32,7 +32,7 @@ export function ConstantsSection() {
             <Divider style={{ height: 1 }} />
           </>
         ) : null}
-        <ConstantItem title="Supported SDKs" value={Environment.supportedSdksString} />
+        <ConstantItem title="Supported SDK" value={Environment.supportedSdksString} />
```
Comment (FiberJW) [inline apps/expo-go/src/screens/SettingsScreen/ConstantsSection.tsx:35]: Ah yes, we only support one SDK in Expo Go now!

File: apps/expo-go/src/components/NavigationScrollView.tsx [hunk 2]
```diff
@@ -18,27 +14,12 @@ type StyledScrollViewProps = PropsWithChildren<
...
   const ref = useRef(null);
-  const { style, ...otherProps } = props;
-  const backgroundColor = useThemeBackgroundColor(props, 'bodyBackground');
 
   useScrollToTop(ref);
 
   const ScrollView = Platform.OS === 'android' ? RNScrollView : RNGHScrollView;
 
-  return <ScrollView style={[{ backgroundColor }, style]} {...otherProps} ref={ref} />;
+  return <ScrollView style={style} {...otherProps} ref={ref} />;
```
Comment (FiberJW) [inline apps/expo-go/src/components/NavigationScrollView.tsx:24]: Were these background color changes tested on Android?

**Review by FiberJW — APPROVED** [review]
This generally looks good, but I'd want you to verify that these changes look as you intended on Android before merging

File: apps/expo-go/src/components/NavigationScrollView.tsx (hunk 2 above)
Comment (Simek) [reply apps/expo-go/src/components/NavigationScrollView.tsx:24]: Just got a test device, and the background color seems to be correct. However, I have pushed the small tweak for the Account "modal".

**Comment by Simek** [comment]
### Android

Screenshot_20240722-222437

Screenshot_20240722-223058

Screenshot_20240722-222759

**Review by FiberJW — APPROVED** [review]
Thanks for verifying that Android renders as intended! Let's ship it!

**Pull request merged**